framework/web/Request.php

```diff
@@ -1366,32 +1358,36 @@ protected function generateCsrfToken()
     }
 
     /**
-     * Returns the XOR result of two strings.
-     * If the two strings are of different lengths, the shorter one will be padded to the length of the longer one.
-     * @param string $token1
-     * @param string $token2
-     * @return string the XOR result
+     * @param string $token An unmasked token.
+     * @return string A masked token.
      */
-    private function xorTokens($token1, $token2)
+    protected function maskCsrfToken($token)
     {
-        $n1 = StringHelper::byteLength($token1);
-        $n2 = StringHelper::byteLength($token2);
-        if ($n1 > $n2) {
-            $token2 = str_pad($token2, $n1, $token2);
-        } elseif ($n1 < $n2) {
-            $token1 = str_pad($token1, $n2, $n1 === 0 ? ' ' : $token1);
-        }
+        // Mask always equal length (in bytes) to token.
+        $mask = Yii::$app->security->generateRandomKey(strlen($token));
```

Comment:
From [the manual](http://php.net/strlen):
````
Note:
strlen() returns the number of bytes rather than the number of characters in a string.
````
(I don't specifically care which method we use, they seem to be equivalent though)